black_it/loss_functions/base.py

```diff
@@ -17,25 +17,35 @@
 """This module defines the 'BaseLoss' base class."""
 
 from abc import ABC, abstractmethod
-from typing import Optional
+from typing import Callable, List, Optional
 
 import numpy as np
 from numpy.typing import NDArray
 
 from black_it.utils.base import assert_
 
+TimeSeriesFilter = Callable[[NDArray[np.float64]], NDArray[np.complex128]]
+"""A filter that receives a time series and returns its filtered version. Used by the BaseLoss constructor."""
+
 
 class BaseLoss(ABC):
     """BaseLoss interface."""
 
-    def __init__(self, coordinate_weights: Optional[NDArray] = None):
+    def __init__(
+        self,
+        coordinate_weights: Optional[NDArray] = None,
+        coordinate_filters: Optional[List[Optional[Callable]]] = None,
```

Comment:
I guess we could have three behaviours of this parameter:

1. None -> the identity function
2. callable -> a filter that will be applied to every dimension
3. list of callables/none -> a filter specified for each dimension

1 and 3 are in place; if useful, we could add 2 (*even after the PR is merged*)

Reply:
Thanks this might be a good idea for a future PR if we find it to be potentially useful. For the moment I would not implement it though

Reply:
Another idea we talked about is to use dictionaries rather than lists.

For example, say we have 10 dimensions, but we want to apply a filter for the 3rd, the 5th and the 7th dimensions. With the current interface we had to do something like:

```python
[None] * 2 + [filter_for_3rd] + [None] + [filter_for_5th] + [None] + [filter_for_7th]  + [None] * 3
```

Instead, if we used dictionaries with dimension indexes as keys, we could do:
```python
{
  2 : filter_for_3rd # (or 3?)
  4 : filter_for_5th # (or 5?)
  6 : filter_for_7th # (or 7?)
}
```